### DISCUSSION TITLE
release版本应用崩溃

### DISCUSSION THREAD

**Discussion by kang12** [Tencent/libpag#1991] [Q&A] [answered]
使用pag 4.3.33版本，生成PAGImage，编译出来的debug使用是正常的，但是编译出来的release就会崩溃，已配置混淆，可以帮忙看下是什么原因吗

java.lang.UnsatisfiedLinkError: No implementation found for void org.libpag.PAGImage.nativeInit() (tried Java_org_libpag_PAGImage_nativeInit and Java_org_libpag_PAGImage_nativeInit__)

12-12 17:17:46.533 16660 16660 F DEBUG   : Revision: '0'
12-12 17:17:46.533 16660 16660 F DEBUG   : ABI: 'arm'
12-12 17:17:46.533 16660 16660 F DEBUG   : pid: 16109, tid: 16593, name: Thread-104 
12-12 17:17:46.538 16660 16660 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
12-12 17:17:46.548 16660 16660 F DEBUG   : Abort message: 'java_vm_ext.cc:534] JNI DETECTED ERROR IN APPLICATION: JNI FindClass called with pending exception java.lang.NoSuchMethodError: no non-static method "Landroidx/exifinterface/media/ExifInterface;.getAttributeInt(Ljava/lang/String;I)I"'
12-12 17:17:46.548 16660 16660 F DEBUG   :     r0 00000000  r1 000040d1  r2 00000006  r3 00000008
12-12 17:17:46.548 16660 16660 F DEBUG   :     r4 00003eed  r5 000040d1  r6 8df8a4bc  r7 0000010c
12-12 17:17:46.548 16660 16660 F DEBUG   :     r8 00000000  r9 aa98c1c0  sl 0000000a  fp 8df8a508
12-12 17:17:46.549 16660 16660 F DEBUG   :     ip 8d56b5c0  sp 8df8a4a8  lr ad860779  pc ad85a49e  cpsr 28070030
12-12 17:17:46.669 16660 16660 F DEBUG   : 
12-12 17:17:46.669 16660 16660 F DEBUG   : backtrace:
12-12 17:17:46.669 16660 16660 F DEBUG   :     #00 pc 0001a49e  /system/lib/libc.so (abort+63)
12-12 17:17:46.669 16660 16660 F DEBUG   :     #01 pc 0035b0f9  /system/lib/libart.so (art::Runtime::Abort(char const*)+392)
12-12 17:17:46.669 16660 16660 F DEBUG   :     #02 pc 0041ac2d  /system/lib/libart.so (android::base::LogMessage::~LogMessage()+452)
12-12 17:17:46.669 16660 16660 F DEBUG   :     #03 pc 0024d659  /system/lib/libart.so (art::JavaVMExt::JniAbort(char const*, char const*)+1212)
12-12 17:17:46.669 16660 16660 F DEBUG   :     #04 pc 0024d7db  /system/lib/libart.so (art::JavaVMExt::JniAbortV(char const*, char const*, std::__va_list)+58)
12-12 17:17:46.669 16660 16660 F DEBUG   :     #05 pc 000d5dd3  /system/lib/libart.so (art::ScopedCheck::AbortF(char const*, ...)+42)
12-12 17:17:46.669 16660 16660 F DEBUG   :     #06 pc 000d5953  /system/lib/libart.so (art::ScopedCheck::CheckThread(_JNIEnv*)+274)
12-12 17:17:46.669 16660 16660 F DEBUG   :     #07 pc 000d42fd  /system/lib/libart.so (art::ScopedCheck::Check(art::ScopedObjectAccess&, bool,

**Comment by kevingpqi123** [comment]
这个问题重点就是看你们 debug 和 release 模式的差异是啥，从堆栈上看，就是相应方法没找到，可能和混淆强相关

**Comment by domchen** [comment, marked as answer]
检擦一下abifilters，以及你运行的目标机器架构，这个一般是架构问题没有匹配上，报错内容是缺少对应架构的so文件。另外如果做了so动态下发，再检擦一下下发逻辑是否成功。
